src/libsyntax/visit.rs

```diff
@@ -163,6 +162,12 @@ pub trait Visitor<'ast>: Sized {
     fn visit_fn_header(&mut self, _header: &'ast FnHeader) {
         // Nothing to do
     }
+    fn visit_field(&mut self, _: &'ast Field) {
+        // Nothing to do
```

Comment:
Methods on the `Visitor` trait must always contain `walk_*` calls inside of them (unless they visit some leaf node without internal structure, like `ast::Name`).

`visit_*` methods must also always be called from their parent `walk_*` methods, otherwise they won't be overridable by specific visitors (which is the purpose of the the `Visitor` trait).